File: src/src/yjhwang/level_2/캐시.js
function solution(cacheSize, cities) {
    const caches = [];
    const CACHE_HIT = 1;
    const CACHE_MISS = 5;
  
    if (cacheSize === 0) {
      return cities.length * CACHE_MISS;
    }
  
    let times = 0;
    for (let i = 0; i < cities.length; i++) {
      const city = String(cities[i]).toLowerCase();
  
      const cacheIndex = caches.findIndex((data) => data.city === city);
      if (cacheIndex > -1) {
        times += CACHE_HIT;
        caches[cacheIndex].time = i;
      } else {
        if (caches.length >= cacheSize) {
          caches.pop();
        }
  
        times += CACHE_MISS;
        caches.push({ city, time: i });
      }
  
      caches.sort((next, prev) => prev.time - next.time);
    }
  
    return times;
  }

File: src/src/bmkim/level_2/캐시.java
static class Solution {
    static final int CACHE_HIT = 1;
    static final int CACHE_MISS = 5;
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;

        if (cacheSize == 0) {
            return cities.length*5;
        }
        LinkedHashSet cache = new LinkedHashSet<>();
        for (int i=0; i< cities.length; i++) {
            String city = cities[i].toUpperCase();

            if(cache.contains(city)) {
                cache.remove(city);
                answer += CACHE_HIT;
            } else {
                if (cache.size() == cacheSize) {
                    Iterator iterator = cache.iterator();
                    iterator.next();
                    iterator.remove();
                }
                answer += CACHE_MISS;
            }
            cache.add(city);
        }
        return answer;
    }
}